apis/v1alpha1/cluster_kprobe_program_types.go

```diff
@@ -19,24 +19,31 @@ package v1alpha1
 
 // ClKprobeProgramInfo contains the information for the kprobe program
 type ClKprobeProgramInfo struct {
-	// The list of points to which the program should be attached.  The list items
-	// are optional and may be udated after the bpf program has been loaded
+	// links is optional and is the list of hook points to which the KProbe program
+	// should be attached. The eBPF program is loaded in kernel memory when the BPF
+	// Application CRD is created and the selected Kubernetes nodes are active. The
+	// eBPF program will not be triggered until the program has also been attached
+	// to a hook point described in this list. Items may be added or removed from
+	// the list at any point, causing the eBPF program to be attached or detached.
+	//
```

Comment:
Can this be a general statement made in the bpf application CRDs that applies to all program types rather than repeat it for each type?  I suggested some additional explanation for Fentry and Fexit that explain how they are a little different, but the general statement still applies to them.

Reply:
As a general comment, I think that all of the details for each program type should be described on the attributes of the associated *AttachInfo structure and not anywhere else.